cmd/ci-operator/main.go

```diff
@@ -1425,27 +1437,35 @@ func eventRecorder(kubeClient *coreclientset.CoreV1Client, authClient *authclien
 		templatescheme.Scheme, coreapi.EventSource{Component: namespace}), nil
 }
 
-func getSSHSecretFromPath(sshKeyPath string) (*coreapi.Secret, error) {
+func getCloneSecretFromPath(cloneAuthType steps.CloneAuthType, secretPath string) (*coreapi.Secret, error) {
 	secret := &coreapi.Secret{Data: make(map[string][]byte)}
-	secret.Type = coreapi.SecretTypeSSHAuth
-
-	b, err := ioutil.ReadFile(sshKeyPath)
+	data, err := ioutil.ReadFile(secretPath)
 	if err != nil {
-		return nil, fmt.Errorf("could not read file %s for secret: %v", sshKeyPath, err)
+		return nil, fmt.Errorf("could not read file %s for secret: %v", secretPath, err)
 	}
+	hash := getHashFromBytes(data)
 
-	hash := sha256.New()
-	hash.Write(b)
-	hashFromBytes := oneWayNameEncoding.EncodeToString(hash.Sum(nil)[:5])
-	secret.Name = fmt.Sprintf("ssh-%s", hashFromBytes)
-
-	// Secret.Data["ssh-privatekey"] is required on SecretTypeSSHAuth type.
-	// https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5466-L5470
-	secret.Data[coreapi.SSHAuthPrivateKey] = b
+	if cloneAuthType == steps.CloneAuthTypeSSH {
+		secret.Name = fmt.Sprintf("ssh-%s", hash)
+		secret.Type = coreapi.SecretTypeSSHAuth
+		// Secret.Data["ssh-privatekey"] is required on SecretTypeSSHAuth type.
+		// https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5466-L5470
+		secret.Data[coreapi.SSHAuthPrivateKey] = data
+	} else if cloneAuthType == steps.CloneAuthTypeOAuth {
+		secret.Type = coreapi.SecretTypeOpaque
+		secret.Name = fmt.Sprintf("oauth-%s", getHashFromBytes(data))
+		secret.Data[steps.OauthSecretKey] = []byte(strings.TrimSpace(string(data)))
```

Comment:
the `TrimSpace` seems misplaced here. IMO, we should `TrimSpace` immediately after we `ReadFile` (even before we hash) and just use the cleaned string later

Reply:
well, I know how that it seems misplaced, but we have already broken prow once because of this. It seems that we need to Trim it otherwise the `ReadFile` will append a newline in the value. 
See [prow/config/secret/secret.go#L45](https://github.com/kubernetes/test-infra/blob/master/prow/config/secret/secret.go#L45)

Reply:
But your concern is correct. I will move this.

Reply:
Yeah, that's what I meant - we need to `TrimSpace`, just not here but earlier.

Reply:
I am not sure if trimming space in the ssh key will cause any problems. I will validate that and then address your comment.

Reply:
Should not hurt. One more thing - `bytes` has `TrimSpace` too, no need to convert it back and forth.

https://golang.org/pkg/bytes/#TrimSpace